### PR TITLE
WRP-7826: Upgraded eslint-plugin-react to v7.32.2 and created new eslint.config.js

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop [stanca.pop@lgepartner.com](mailto:stanca.pop@lgepartner.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] I have run automated testing and it is passed
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
_eslint-plugin-react_ v7.32.0 has been released on Jan 11, 2023. This version has several breaking changes, including jsx enable config being moved under languageOptions.
This new schema change cannot work on the .eslintrc.js file, only works on the eslint.config.js file.  
`.eslintrc.js` will be deprecated from eslint v9, only `eslint.config.js` file will be supported.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Upgraded _eslint-plugin-react_ and replaced `.eslintrc.js` with `eslint.config.js`. Several eslint disable directives were deleted as Travis errored with message that they were useless.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-7826

### Comments

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]
Travis build errored, but this seems to be not related to the code in the PR

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		plugins: {
+			enactConfig
+		},
+		rules: {
+			'max-nested-callbacks': 'off',
+			'no-console': 'off',
+			'react/forbid-foreign-prop-types': 'off' // proptypes not removed in storybook config
+		}
+	}
+];
```
Comment (hong6316) [inline eslint.config.js:30]: Could you explain the changed config file in more detail in "Resolution"?
I compared these configurations with the deleted .eslintrc.js, and there seem to be many changes.

File: package.json [hunk 2]
```diff
@@ -48,7 +48,8 @@
     "eslint-plugin-jest": "^27.2.1",
     "eslint-plugin-jsx-a11y": "^6.7.1",
     "eslint-plugin-prettier": "^4.2.1",
-    "eslint-plugin-react": "7.31.11",
-    "eslint-plugin-react-hooks": "^4.6.0"
+    "eslint-plugin-react": "^7.32.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "globals": "^13.20.0"
```
Comment (hong6316) [inline package.json:53]: is that needed?

Comment (stanca-pop-lgp) [reply package.json:53]: It was, if we keep the implementation like here: https://github.com/jsx-eslint/eslint-plugin-react#plugin.

File: eslint.config.js (hunk 1 above)
Comment (hong6316) [reply eslint.config.js:28]: ok...hope the travis works...
eslintrc -> eslint.config.js differences are little bit complicated.
Please check and share to us, I also mention & check together.

**Pull request closed without merging**